File: packages/python/src/output_guard/batch.py
"""batch: Light list-comp wrappers over ``validate`` and ``repair``.

Mirror of packages/ts/src/batch.ts.
"""

from __future__ import annotations

from typing import Any

from .adapters.pydantic_ import SchemaAdapter
from .errors import ParseError, SchemaValidationError
from .orchestrator import SchemaValidator, repair, validate
from .results import Format, RepairResult, ValidationResult


def validate_batch(
    items: list[str],
    fmt: Format = "auto",
    schema: SchemaValidator | SchemaAdapter[Any] | None = None,
) -> list[ValidationResult]:
    return [validate(item, fmt, schema) for item in items]


def repair_batch(
    items: list[str],
    fmt: Format = "auto",
    schema: SchemaValidator | SchemaAdapter[Any] | None = None,
) -> list[RepairResult | dict[str, str]]:
    results: list[RepairResult | dict[str, str]] = []
    for item in items:
        try:
            results.append(repair(item, fmt, schema))
        except (ParseError, SchemaValidationError) as e:
            results.append({"error": str(e)})
    return results

File: packages/ts/src/batch.ts
/**
 * batch: Light array-map wrappers over validate and repair.
 */
import type { Format, RepairResult, ValidationResult } from "./results.js";
import { validate, repair } from "./orchestrator.js";

type SchemaValidator<T = unknown> = (data: unknown) => ValidationResult<T>;

export const validateBatch = <T = unknown>(
  items: string[],
  format: Format = "auto",
  schema?: SchemaValidator<T>,
): ValidationResult<T>[] => items.map((item) => validate<T>(item, format, schema));

export const repairBatch = <T = unknown>(
  items: string[],
  format: Format = "auto",
  schema?: SchemaValidator<T>,
): Array<RepairResult<T> | { error: string }> =>
  items.map((item) => {
    try {
      return repair<T>(item, format, schema);
    } catch (e) {
      return { error: String(e) };
    }
  });
